systemd/resolved/resolved.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolved
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/godbus/dbus/v5"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const namespace = "systemd_resolved"
+
+type Collector struct {
+	ctx                         context.Context
+	logger                      log.Logger
+	resolvedCurrentTransactions *prometheus.Desc
+	resolvedTotalTransactions   *prometheus.Desc
+	resolvedCurrentCacheSize    *prometheus.Desc
+	resolvedTotalCacheHits      *prometheus.Desc
+	resolvedTotalCacheMisses    *prometheus.Desc
+	resolvedTotalSecure         *prometheus.Desc
+	resolvedTotalInsecure       *prometheus.Desc
+	resolvedTotalBogus          *prometheus.Desc
+	resolvedTotalIndeterminate  *prometheus.Desc
```

Comment:
We've been moving towards a pattern where these vars are package level, rather than part of the `Collector` struct.

This makes them declared once at startup time, instead of every time `NewCollector()` is called.

Reply:
Thank you, fixed at https://github.com/prometheus-community/systemd_exporter/pull/119/commits/680cf8159ed35caaea315dda5809e8bfe8f355f1

Could you please check it again?